src/main/kotlin/io/github/mojira/arisa/modules/PrivacyModule.kt

```diff
@@ -12,7 +12,7 @@ class PrivacyModule(
 ) : Module {
     private val patterns: List<Regex> = listOf(
         """\(Session ID is token:""".toRegex(),
-        """(?<!\[~)\b[a-zA-Z0-9.-_]+@[a-zA-Z0-9.-_]+\.[a-zA-Z0-9.-_]{2,}\b(?!])""".toRegex(),
+        """(?<!\[~)\b[a-zA-Z0-9.-_]+@[a-zA-Z0-9.-_]+\.[a-zA-Z0-9.-_]{2,}""".toRegex(),
```

Comment:
This is kinda simple, we probably should work on a regex that only accepts valid TLDs at the end and use it on the privacy module too. @CubeTheThird maybe you have some ideas?

Reply:
I might be missing some context, but the original pattern before the edit here doesn't allow for [ or ~ at the start, nor ] at the end. You've removed the check for the ] at the end, but the checks at the start are still present. What exactly is the goal then? Also, if you're looking at refining the detection of valid email addresses, there's a lot of info about that subject on https://emailregex.com/, but that's a bit of a separate conversation.

Reply:
The goal of the regex is to match emails that should be hidden:
- `example@domain.com`
- `[email me|mailto:example@domain.com]`

But not emails in user mentions:
- `[~example.domain.com]`

Reply:
It we are to improve the regex further, there are two open issues that could be considered: #500 and #510.

Reply:
I added #500 by limiting the number of charachters in the TLD and not matching with underscores according to RFC 1035 (underscores are not allowed in domain names)